### DISCUSSION TITLE
How to compose a view with search header

### DISCUSSION THREAD

**Discussion by kuacci** [microsoft/vscode-discussions#891] [Extension Development QnA] [answered]
I am developing an extension which needs the search and list. The idea UI likes the built-in Extension view of vs code. May I know if it is possible to create a view just like this one?
![image](https://github.com/microsoft/vscode-discussions/assets/5857515/9f53b545-b354-44c1-b537-2dbd14b4d972)

**Comment by alefragnani** [comment]
Hi @kuacci ,

To accomplish similar UI, you will need to use [Webview API](https://code.visualstudio.com/api/extension-guides/webview), as there is no API to add Inputs/Search at the top of Tree View, nor such complex (visually speaking) List/TreeViews.

But, if the current TreeView API works for you, without the need for a custom list like the screenshot, and the native `Tree Find Control` works for you, you could use it as well. More details here https://github.com/microsoft/vscode/issues/146806#issuecomment-1122758195

Hope this helps

**Reply by kuacci** [reply]
@alefragnani 
Thanks for your reply. I've tried `Tree Find Control` with `TreeView`, it works when the data set is small. But it might be a problem when the data set is huge, I mean the tree item might reach to 10k+ . 

I not sure if it is possible to compose webview and TreeView together within a parent view?

-Kuacci.

**Reply by alefragnani** [reply, marked as answer]
No, you can't compose Webview and TreeView. You must choose one.

I guess your best bet would be Webviews, as you will have more flexibility for your requirements. And, if you intend to follow VS Code design language, take a look at the [Webview UI Toolkit for Visual Studio Code](https://github.com/microsoft/vscode-webview-ui-toolkit), as it could help you.

Hope this helps

**Reply by kuacci** [reply]
Thanks for your suggestion. I guess I have to create such view by myself in Webview.
